internal/monitors/cpu/cpu.go

```diff
@@ -0,0 +1,245 @@
+package cpu
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+	"time"
+
+	"github.com/shirou/gopsutil/cpu"
+	"github.com/signalfx/golib/datapoint"
+	"github.com/signalfx/signalfx-agent/internal/core/common/constants"
+	"github.com/signalfx/signalfx-agent/internal/core/config"
+	"github.com/signalfx/signalfx-agent/internal/monitors"
+	"github.com/signalfx/signalfx-agent/internal/monitors/types"
+	"github.com/signalfx/signalfx-agent/internal/utils"
+	"github.com/signalfx/signalfx-agent/internal/utils/gopsutilhelper"
+	log "github.com/sirupsen/logrus"
+)
+
+const monitorType = "cpu"
+const cpuUtilName = "cpu.utilization"
+const percoreMetricName = "cpu.utilization_per_core"
+
+// setting cpu.Times to a package variable for testing purposes
+var times = cpu.Times
+
+// MONITOR(cpu):
+// This monitor reports cpu and cpu utilization metrics.
+//
+//
+// ```yaml
+// monitors:
+//  - type: cpu
+// ```
+
+var logger = log.WithFields(log.Fields{"monitorType": monitorType})
+
+type totalUsed struct {
+	Total float64
+	Used  float64
+}
+
+// TODO: make ProcFSPath a global config
+
+// Config for this monitor
+type Config struct {
+	config.MonitorConfig `singleInstance:"true" acceptsEndpoints:"false"`
+	// The path to the proc filesystem. Useful to override in containerized
+	// environments.  (Does not apply to windows)
+	ProcFSPath      string `yaml:"procFSPath" default:"/proc"`
+	previousPerCore map[string]*totalUsed
```

Comment:
Put this on the monitor struct, not the config struct.  Config struct is hashed to determine changes and this will break it.

Reply:
ack